framework/sandstone.cpp

```diff
@@ -2685,6 +2706,121 @@ static void warn_deprecated_opt(const char *opt)
             program_invocation_name, opt);
 }
 
+
+static bool system_is_idle(float idle_threshold)
+{
+    #ifdef __linux__
+
+    FILE *loadavg;
+    float load_1, load_5, load_15;
+    int ret;
+
+    // Look at loadavg for hints whether the system is reasonably idle or not
+    // any error and we assume it's busy/under load for simplicity
+
+    loadavg = fopen("/proc/loadavg", "r");
+    if (!loadavg)
+        return false;
+
+    ret = fscanf(loadavg, "%f %f %f", &load_1, &load_5, &load_15);
+    fclose(loadavg);
+
+    if (ret != 3)
+       return false;
+
+    if (load_5 <= idle_threshold)
+       return true;
+    #else //__linux__
+    return true;//TO DO implement idle trigger for Windows
+    #endif
+
+    return false;
+}
+
+static void background_scan_init(void)
+{
+    MonotonicTimePoint now = MonotonicTimePoint::clock::now();
+
+    // init timestamps to 1day old - this quickstarts testing on first run
+    if ((sApp->background_scan.init_timestamp == true) && (sApp->service_background_scan == true)) {
+        for (auto i = 0; i < sApp->background_scan.number_of_timestamps; i++) {
+            MonotonicTimePoint timestamp_init_val = 
+                    (now - sApp->background_scan.day_delay_between_tests);
+            sApp->background_scan.timestamp[i] = timestamp_init_val;
+        }
+    }
+}
+
+static void background_scan_update_load_threshold(void)
+{
+    MonotonicTimePoint now = MonotonicTimePoint::clock::now();
+
+    int64_t time_from_last_test = 
+        ((now - sApp->background_scan.timestamp[sApp->background_scan.timestamp_newest]) /
+        sApp->background_scan.number_seconds_in_hour).count();
```

Comment:
If you just wanted the number of hours since the last test:
```c++
    hours time_since_last_test = duration_cast<hours>(now - last_timestamp);
```
And below you use `hours.count()`.

3599.999999999 seconds since the last timestamp will still be zero hours.